### PR TITLE
[TS] LPS-148758 Set extension based on file's mimeType if extension is empty

### PR DESCRIPTION
**Pull request by evibudai**
Hi Team,
@robertoDiaz

LPS: https://issues.liferay.com/browse/LPS-148758

When we open a link (Latest Version URL) of a file that has no extension in its filename (ex.: "test", instead "test.pdf"), the logic sets the disposition type to “attachment” so it gets downloaded.
Fix: When the file has no extension in its name we use the contentType as a fallback to determine the extension. Result: the file is displayed correctly when opening its URL.

Could you please review it?
Thank you!

### CODE REVIEW

**Comment by evibudai** [comment]
ci:test

**Comment by evibudai** [comment]
ci:test:relevant

**Comment by peterpetrekanics** [comment]
Dear @AliciaGarciaGarcia ,

May I ask for your kind help?
On this pull request we have received failures, but I am not sure if they are related to the code changes that were made.
Is there a way to tell for sure what exactly triggered the failures?
Many thanks,
Peter

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
Hi @evibudai @peterpetrekanics 

It's failing the SF. 
Can you fix it and resend?

Thanks!

File: portal-kernel/src/com/liferay/portal/kernel/servlet/ServletResponseUtil.java [hunk 1]
```diff
@@ -481,6 +481,11 @@ protected static void setHeaders(
 			String extension = GetterUtil.getString(
 				FileUtil.getExtension(fileName));
 
+			if (extension.isEmpty()) {
+				extension = MimeTypesUtil.getExtensions(contentType)
+					.iterator().next().substring(1);
```
Comment (AliciaGarciaGarcia) [inline portal-kernel/src/com/liferay/portal/kernel/servlet/ServletResponseUtil.java:486]: Source Formatter failing.

**Review by AliciaGarciaGarcia — CHANGES_REQUESTED** [review]

**Comment by IstvanD** [comment]
Hi @AliciaGarciaGarcia, @evibudai is out of office, so I took over her work in the LPS.
I have sent a new pull request with source formatting corrections.
https://github.com/liferay-lima/liferay-portal/pull/2828

**Comment by AliciaGarciaGarcia** [comment]
https://github.com/liferay-lima/liferay-portal/pull/2828

**Pull request closed without merging**